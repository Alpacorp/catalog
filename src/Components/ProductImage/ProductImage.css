.product-image {
  width: auto;
}

.product-image img {
  width: 100%;
  max-width: 250px;
  max-height: 180px;
  border-radius: 5px;
  object-fit: cover;
}

@media screen and (max-width: 750px) {
  .product-image {
    width: auto;
  }

  .product-image figure {
    margin: 10px;
  }

  .product-image img {
    width: 100%;
    height: auto;
  }
}
