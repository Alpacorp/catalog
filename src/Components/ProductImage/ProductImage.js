import PropTypes from "prop-types";
import "./ProductImage.css";

const ProductImage = ({ id, image, name, styleImg }) => {
  return (
    <>
      <div className={styleImg} key={id + "image"} tabIndex="0">
        <figure>
          <img src={image} alt={name} />
        </figure>
      </div>
    </>
  );
};

ProductImage.propTypes = {
  id: PropTypes.number,
  image: PropTypes.any,
  name: PropTypes.string,
  styleImg: PropTypes.string,
};

export default ProductImage;
