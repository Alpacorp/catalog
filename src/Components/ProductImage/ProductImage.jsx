import PropTypes from "prop-types";
import "./ProductImage.css";

const ProductImage = ({ id, image, name, styleImg }) => {
  return (
    <>
      <div
        data-testid="test-div-img"
        className={styleImg}
        key={id + "image"}
        tabIndex="0"
        itemScope
        itemType="http://schema.org/Thing"
      >
        <figure>
          <img itemProp="image" src={image} alt={name} />
        </figure>
      </div>
    </>
  );
};

ProductImage.propTypes = {
  id: PropTypes.number.isRequired,
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  styleImg: PropTypes.string,
};

export default ProductImage;
