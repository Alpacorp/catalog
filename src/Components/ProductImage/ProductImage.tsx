import "./ProductImage.css";

interface ProductImageProps {
  id: number;
  image: string;
  name: string;
  styleImg?: string;
}

const ProductImage = ({ id, image, name, styleImg }: ProductImageProps) => {
  return (
    <>
      <div
        data-testid="test-div-img"
        className={styleImg}
        key={id + "image"}
        tabIndex={0}
        itemScope
        itemType="http://schema.org/Thing"
      >
        <figure>
          <img itemProp="image" src={image} alt={name} />
        </figure>
      </div>
    </>
  );
};

export default ProductImage;
