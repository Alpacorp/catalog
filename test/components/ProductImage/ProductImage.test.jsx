import { render, screen } from "@testing-library/react";
import ProductImage from "../../../src/Components/ProductImage/ProductImage";

describe("Test in <ProductImage />", () => {
  const id = 1;
  const image = "https://via.placeholder.com/150";
  const name = "Product 1";
  const styleImg = "product-image";

  test("should render to match snapshot", () => {
    const { container } = render(
      <ProductImage id={id} image={image} name={name} styleImg={styleImg} />
    );
    expect(container).toMatchSnapshot();
  });

  test("should render image with SRC and ALT attributes", () => {
    render(
      <ProductImage id={id} image={image} name={name} styleImg={styleImg} />
    );
    expect(screen.getByRole("img").src).toBeTruthy();
    expect(screen.getByRole("img").alt).toBeTruthy();
  });

  test("should render to match snapshot main DIV with styleImg", () => {
    render(
      <ProductImage id={id} image={image} name={name} styleImg={styleImg} />
    );
    expect(screen.getByTestId("test-div-img").className).toBeTruthy();
  });
});
