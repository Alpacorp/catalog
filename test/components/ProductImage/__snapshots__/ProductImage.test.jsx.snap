// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <ProductImage /> should render to match snapshot 1`] = `
<div>
  <div
    class="product-image"
    data-testid="test-div-img"
    tabindex="0"
  >
    <figure>
      <img
        alt="Product 1"
        src="https://via.placeholder.com/150"
      />
    </figure>
  </div>
</div>
`;
